### PR TITLE
Improved helper function that waits for podsets to be ready.

### PR DESCRIPTION
**Pull request by greyerof**
The helper function WaitForAllPodsetReady was taking too long in case there were too many deployments+statefulsets to check. In that case, as it was waiting sequentially for each of them to time out (7mins), the test case lifecycle-pod-recreation could take hours to finish in the worst case scenario (none of the podsets are ready).

I've refactored it to check the status of every podset in every pass and wait 15secs before checking again the not-ready ones. There's no per-podset timeout, just a global timeout that needs to be passed for all of them to be ready.

### CODE REVIEW

File: cnf-certification-test/lifecycle/podsets/podsets.go [hunk 1]
```diff
@@ -91,26 +92,125 @@ func WaitForStatefulSetReady(ns, name string, timeout time.Duration) bool {
...
+	return stsInfo
+}
+
+// Helper function that checks the status of each deployment in the slice and returns
+// a slice with the not-ready ones.
+func getNotReadyDeployments(deployments []*provider.Deployment) []*provider.Deployment {
+	notReadyDeployments := []*provider.Deployment{}
+	for _, dep := range deployments {
+		ready, err := isDeploymentReady(dep.Name, dep.Namespace)
+		if err != nil {
```
Comment (sebrandon1) [inline cnf-certification-test/lifecycle/podsets/podsets.go:144]: We should also log the contents of `err` here.

**Review by sebrandon1 — COMMENTED** [review]
Just some initial comments.

**Review by bnshr — CHANGES_REQUESTED** [review]

**Review by sebrandon1 — APPROVED** [review]
Looks good.  We could also add unit tests around the funcs in `cnf-certification-test/lifecycle/podsets/podsets.go`.

File: cnf-certification-test/lifecycle/podsets/podsets.go [hunk 2]
```diff
@@ -91,26 +92,125 @@ func WaitForStatefulSetReady(ns, name string, timeout time.Duration) bool {
...
+func WaitForAllPodSetsReady(env *provider.TestEnvironment, timeout time.Duration) (claimsLog loghelper.CuratedLogLines, atLeastOnePodsetNotReady bool) {
+	const queryInterval = 15 * time.Second
+
+	deploymentsToCheck := env.Deployments
+	statefulSetsToCheck := env.StatefulSets
+
+	logrus.Infof("Waiting %s for %d podsets to be ready.", timeout, len(deploymentsToCheck)+len(statefulSetsToCheck))
+	for startTime := time.Now(); time.Since(startTime) < timeout; {
+		logrus.Infof("Checking Deployments readiness of Deployments %v", getDeploymentsInfo(deploymentsToCheck))
+		deploymentsToCheck = getNotReadyDeployments(deploymentsToCheck)
```
Comment (edcdavid) [inline cnf-certification-test/lifecycle/podsets/podsets.go:193]: deploymentsToCheck -> notReadyDeployments
statefulSetsToCheck -> notReadyStatefulsets

**Review by edcdavid — APPROVED** [review]

**Pull request merged**